arangodb/content.md

```diff
@@ -31,32 +31,48 @@ In order to start an ArangoDB instance, run:
 unix> docker run -e ARANGO_RANDOM_ROOT_PASSWORD=1 -d --name arangodb-instance %%IMAGE%%
 ```
 
-Will create and launch the %%IMAGE%% docker instance as background process. The Identifier of the process is printed. By default ArangoDB listen on port 8529 for request and the image includes `EXPOSE 8529`. If you link an application container it is automatically available in the linked container. See the following examples.
+On macOS/ARM, the processor architecture chosen by default may be x86-64 (`linux/amd64`), which is not optimal. You can pass the `--platform` flag to the `docker run` command to specify the appropriate operating system and architecture for the container (`linux/arm64/v8`):
```

Comment:
> On macOS/ARM, the processor architecture chosen by default may be x86-64 (`linux/amd64`), which is not optimal.

:grimacing: this is definitely a bug -- do you have more details for reproducing?

Reply:
The co-worker who had trouble with her M1 device isn't with the company anymore. I would need to find someone else with Apple Silicon for testing to get more details. This may have been fixed since, either on our end or in Docker, I don't know. Choosing `linux/amd64` despite ARM hardware isn't wrong generally speaking because there is Rosetta, but using ARM natively should perform better and I believe you can also run into segfaults with ArangoDB in particular because the x86-64 builds use instructions that Rosetta can't emulate (AVX), leading to SIGILL.

Reply:
What I mean is that on an M1 that's properly configured, it _will_ prefer `linux/arm64/v8`, not `amd64`, and if it isn't, that's a definite bug (in the Docker Engine, Docker Desktop, or the environment itself being configured incorrectly).

Reply:
How can it be (mis-)configured to pick `amd64`, assuming it's not a bug? Are there multiple ways to affect the default platform choice?

Reply:
Yes -- setting `DOCKER_DEFAULT_PLATFORM` in your local environment (or, for example, following a guide that suggests putting it into your `.bashrc`, which is a common problem I've seen) will override Docker's built-in automatic detection.

Reply:
A coworker tried `echo $DOCKER_DEFAULT_PLATFORM; docker run --rm busybox:uclibc arch` on Apple silicon. The env var wasn't set and the selected architecture was `aarch64`. I rephrased the remark accordingly as the automatic selection seems to work as it should. I suppose the other coworker had an env var override, which is probable because of some tooling she used that originally supported x86 only.